### DISCUSSION TITLE
What Crazyflie firmware is supported by the Javalib?

### DISCUSSION THREAD

**Discussion by tallner** [bitcraze/discussions#286] [Q&A]
What firmware versions are supported by the Javalib?

**Comment by tallner** [comment]
I just upgraded to the latest firmware using the python cfclient.
After the update the examples in the javalib does not work anymore.
Is the javalib only supporting a certain firmware?

However in the python client it works and I can use the commands to take of and hover etc.
But I got some messages in the Console when doing propeller tests
STAB: WARNING: stabilizer loop rate is off (950)
ESTKALMAN: WARNING: Kalman prediction rate low (96)
HEALTH: Acc noise floor variance X+Y:0.003593, (Z:0.992645)
HEALTH: Motor M1 variance X+Y: 0.24 (Z:0.06), voltage sag:0.16
HEALTH: Motor M2 variance X+Y: 0.92 (Z:0.19), voltage sag:0.17
HEALTH: Motor M3 variance X+Y: 0.65 (Z:0.18), voltage sag:0.17
HEALTH: Motor M4 variance X+Y: 1.70 (Z:0.24), voltage sag:0.17
ESTKALMAN: WARNING: Kalman prediction rate low (24)
ESTKALMAN: State out of bounds, resetting

**Reply by fredg02** [reply]
I have not tested the java lib with recent firmwares, so there might be breaking changes.

**Comment by tallner** [comment]
Regarding the javalib issue, this is the error messages I get when tryin to run the ParamExample
![image](https://user-images.githubusercontent.com/79505011/191906631-72c0627d-454a-457f-baff-d881f62ffeed.png)

**Reply by fredg02** [reply]
I'll try to take a look at it next week. Let's continue the conversation in the [issue you created](https://github.com/fredg02/se.bitcraze.crazyflie.lib/issues/5).

**Comment by tallner** [comment]
Regarding the firmware vs javalib.
I tried the different fw versions and it seems 2021.01 is the latest that can be used.

But when using the 2021.01 fw the python cfclient does no give any information in the console when connecting and the Command Based Flight Control is greyed out.

**Comment by knmcguire** [comment]
@fredg02 could you help out here? or @tallner could you make an issue on the github repo? (https://github.com/fredg02/se.bitcraze.crazyflie.lib).

**Reply by tallner** [reply]
Just created an issue for the javalib vs firmware issue.

I had a couple of questions in this thread, should I start a new one for each question?
In this case I would like to know how the hover mode was implemented in the python client, since I want to create a similar mode in my JavaFX client.
Are there any use cases and how to set the parameters in order to fulfil the use case?
It is sometimes a bit hard to know what the parameters are used for(in a use case), and sometimes i assume several parameters needs to be used or changed.

Cheers
Chris

**Reply by krichardsson** [reply]
Thanks for creating an issue!

Yes, please create a new thread for each question. Thanks!

**Reply by tallner** [reply]
Alright then I'll create a separate regarding use cases. Thanks!
